tests/components/wallbox/test_init.py

```diff
@@ -31,135 +35,118 @@
 )
 
 
-async def test_wallbox_setup_entry(hass: HomeAssistant):
-    """Test Wallbox Setup."""
-    with requests_mock.Mocker() as m:
-        m.get(
-            "https://api.wall-box.com/auth/token/user",
-            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":200}',
-            status_code=200,
-        )
-        m.get(
-            "https://api.wall-box.com/chargers/status/12345",
-            text='{"Temperature": 100, "Location": "Toronto", "Datetime": "2020-07-23", "Units": "Celsius"}',
-            status_code=200,
-        )
-        assert await wallbox.async_setup_entry(hass, entry)
-
-    with requests_mock.Mocker() as m, raises(ConnectionError):
-        m.get(
-            "https://api.wall-box.com/auth/token/user",
-            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":404}',
-            status_code=404,
-        )
-        assert await wallbox.async_setup_entry(hass, entry) is False
-
-
 async def test_wallbox_unload_entry(hass: HomeAssistant):
     """Test Wallbox Unload."""
-    hass.data[DOMAIN] = {"connections": {entry.entry_id: entry}}
-
-    assert await wallbox.async_unload_entry(hass, entry)
 
-    hass.data[DOMAIN] = {"fail_entry": entry}
+    await setup_integration(hass)
 
-    with pytest.raises(KeyError):
-        await wallbox.async_unload_entry(hass, entry)
+    assert await hass.config_entries.async_unload(entry.entry_id)
 
 
 async def test_get_data(hass: HomeAssistant):
     """Test hub class, get_data."""
 
-    station = ("12345",)
-    username = ("test-username",)
-    password = "test-password"
+    await setup_integration(hass)
 
-    hub = wallbox.WallboxHub(station, username, password, hass)
-
-    with requests_mock.Mocker() as m:
-        m.get(
+    with requests_mock.Mocker() as mock_request:
+        mock_request.get(
             "https://api.wall-box.com/auth/token/user",
             text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":200}',
             status_code=200,
         )
-        m.get(
-            "https://api.wall-box.com/chargers/status/('12345',)",
-            json=test_response,
+        mock_request.get(
+            "https://api.wall-box.com/chargers/status/12345",
+            json=test_response_rounding_error,
             status_code=200,
         )
-        assert await hub.async_get_data()
+
+        wallbox = hass.data[DOMAIN][CONF_CONNECTIONS][entry.entry_id]
```

Comment:
Please don't interact with any integration details. We'll test the data fetching by setting up the integration and asserting that correct state is present in the state machine.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
These tests turned out to be redundant in the end, by testing the sensor and the config flow this part of init was already being tested. I removed this part therefore.